dpsim-models/src/DP/DP_Ph1_VoltageSource.cpp

```diff
@@ -31,12 +31,20 @@ SimPowerComp<Complex>::Ptr DP::Ph1::VoltageSource::clone(String name) {
 }
 
 void DP::Ph1::VoltageSource::setParameters(Complex voltageRef, Real srcFreq) {
-  auto srcSigSine = Signal::SineWaveGenerator::make(**mName + "_sw");
-  srcSigSine->mVoltageRef->setReference(mVoltageRef);
-  srcSigSine->mFreq->setReference(mSrcFreq);
-  srcSigSine->setParameters(voltageRef, srcFreq);
+  if (srcFreq == 0.0) {
```

Comment:
Maybe it would be better to add another overload of setParameters without frequency. Then, we could avoid the comparison with 0.0. Comparison with a float looks dangerous if it involves equal.

Reply:
Frequency set to 0.0 is also a default parameter of this method. I am afraid that this affects existing examples. I need to check this more thoroughly.

Reply:
I changed this so old code is not affected by my changes.

Reply:
Thanks!